homeassistant/components/amberelectric/sensor.py

```diff
@@ -0,0 +1,246 @@
+
+        for interval in intervals:
+            datum = {}
+            datum["duration"] = interval.duration
+            datum["date"] = interval.date.isoformat()
+            datum["nem_date"] = interval.nem_time.isoformat()
+            datum["per_kwh"] = round(interval.per_kwh)
+            if interval.channel_type == ChannelType.FEED_IN:
+                datum["per_kwh"] = datum["per_kwh"] * -1
+            datum["spot_per_kwh"] = round(interval.spot_per_kwh)
+            datum["start_time"] = interval.start_time.isoformat()
+            datum["end_time"] = interval.end_time.isoformat()
+            datum["renewables"] = round(interval.renewables)
+            datum["spike_status"] = interval.spike_status.value
+
+            if interval.range is not None:
+                datum["range_min"] = interval.range.min
+                datum["range_max"] = interval.range.max
+
+            data["forecasts"].append(datum)
+
+        data[ATTR_ATTRIBUTION] = ATTRIBUTION
+        return data
+
+
+class AmberGridSensor(CoordinatorEntity, SensorEntity):
+    """Sensor to show single grid specific values."""
+
+    def __init__(
+        self,
+        coordinator: AmberUpdateCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize the Sensor."""
+        super().__init__(coordinator)
+        self.site_id = coordinator.site_id
+        self.entity_description = description
+
+    @property
+    def unique_id(self) -> str | None:
+        """Return a unique id for each sensors."""
+        return f"{self.site_id}-{self.entity_description.key}"
+
+    @property
+    def icon(self) -> str | None:
+        """Return the icon of the sensor."""
+        return self.entity_description.icon
+
+    @property
+    def native_value(self) -> str | None:
+        """Return the value of the sensor."""
+        return self.coordinator.data["grid"][self.entity_description.key]
+
+    @property
+    def device_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return additional pieces of information about the sensor."""
+        data = {}
+        data[ATTR_ATTRIBUTION] = ATTRIBUTION
+        return data
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a config entry."""
+    coordinator: AmberUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    current: dict[str, CurrentInterval] = coordinator.data["current"]
+    forecasts: dict[str, list[ForecastInterval]] = coordinator.data["forecasts"]
+
+    entities: list = []
+    for channel_type in current:
+        description = SensorEntityDescription(
+            key="current",
+            name=f"{entry.title} - {friendly_channel_type(channel_type)} Price",
+            native_unit_of_measurement="AUD",
+            device_class=DEVICE_CLASS_MONETARY,
+            state_class=STATE_CLASS_MEASUREMENT,
+            icon=ICONS[channel_type],
```

Comment:
You're not allowed to override the icon of entities with a device class. 
```suggestion
```

Reply:
Removed device class